### DISCUSSION TITLE
Handling @urql/exchange-populate

### DISCUSSION THREAD

**Discussion by zachasme** [dotansimha/graphql-code-generator#5083] [Q&A]
Hi there!

URQL can [automatically populate mutations](https://formidable.com/open-source/urql/docs/advanced/auto-populate-mutations/) using the `@populate` directive:

```graphql
mutation CreateProduct($input: CreateProductInput!) {
  createProduct(input: $input) {
    product @populate
  }
}
```

but it breaks our codegen setup: `GraphQLDocumentError: Field "product" of type "Product!" must have a selection of subfields. Did you mean "product { ... }"?`

Is it possible to ignore fields with the `@populate` directive? We still wish to generate typescript types for mutation inputs.

**Comment by dotansimha** [comment]
Interesting. @kitten @JoviDeCroock can you please share an overview of how this works? Just spread the entire selection set? how does it works? Thanks!

**Reply by JoviDeCroock** [reply]
The populate exchange tracks the fields that are currently queried for a certain mutation.

So let's say we query `todos { id name }` then it will automatically say `updateTodo { id name }`. If we add another observed query with for instance `todos { id name createdAt }` then this would be `updateTodo { id name createdAt }`.

This is how it currently behaves, we are currently expanding on this behavior with: https://github.com/FormidableLabs/urql/issues/964 which would incorporate populating the viewer field and another one that would create queries.
